tests/rules/property.js

```diff
@@ -664,5 +669,15 @@ eslintTester.run("property", rule, {
             ],
             parser: require.resolve("../parsers/fantasy-operator"),
         },
+        {
+            code: "let c; a.innerHTML = `${c}`;",
```

Comment:
With the change above, this test should then be added to the list of `valid` above.

Reply:
Let's also add one that also have a if...else just to be sure that is still marked as disallowed when appropriate:

```js
        // Uninitialized variable declaration.
        {
            code: `
              let c;
              if (cond) { 
                c = '<b>safe</b>';
              } else {
                c = evil;
              }
              a.innerHTML = \`\${c}\`;
            `,
            errors: [
                {
                    message: /Unsafe assignment to innerHTML/,
                    type: "AssignmentExpression"
                }
            ],
            parserOptions: { ecmaVersion: 6 }
        },
```

And maybe also one to make sure this is still marked as unsafe too:

```js
        {
            code: "let c; c = 'apparently-safe'; functionCall(c); n.innerHTML = c.property;",
            parserOptions: { ecmaVersion: 6 },
            errors: [
                {
                    message: /Unsafe assignment to innerHTML/,
                    type: "AssignmentExpression"
                }
            ],
        }
```

We may also add one or two similar tests to tests/rules/method.js to cover also the method calls explicitly:
- for an allowed case:
```js
        // let without initialization.
        {
            code: "let c; n.insertAdjacentHTML('beforebegin', c)",
            parserOptions: { ecmaVersion: 6 }
        },

```
- and a disallowed one:
```js
        {
            code: `
              let c;
              if (cond) { 
                c = '<b>safe</b>';
              } else {
                c = evil;
              }
              n.insertAdjacentHTML('beforebegin', \`\${c}\`);
            `,
            errors: [
                {
                    message: /Unsafe call to n.insertAdjacentHTML for argument 1/,
                    type: "CallExpression"
                }
            ],
            parserOptions: { ecmaVersion: 6 }
        },
```